public/js/NeighbourPlot.js

```diff
@@ -59,103 +62,167 @@ function NeighbourPlot(sampleData, element) {
     this.width = this.fullWidth - this.margin.left - this.margin.right;
     this.height = this.fullHeight - this.margin.top - this.margin.bottom;
 
-    this.destroy();
-    this.drawScatterplot();
+    this.draw();
 }
 
 /**
- * drawScatterplot: Draw the scatterplot.
+ * applyFilterStyling: Update styling of points according to filter status.
  *
- * Draw the SVG canvas, axis, scatterplot points and tooltips.
+ * Update the class of scatterplot points, and bring unfiltered points
+ * to the top of the SVG stack.
  *
  * @this {NeighbourPlot}
+ * @param {array} newData - The new dataset to display.
```

Comment:
e.g. give a JSON example in the docstring, and what is expected to happen when that example is passed.